### ISSUE TITLE
Different optimization level for a height range?

### ISSUE THREAD

**Issue by klauskobald** [Whinarn/UnityMeshSimplifier#50]
I would like to have the mesh (terrain) being detailed at a certain elevation level say -1 to +2 meters. This would create a better beach. On the other side I would like to have a highly optimised mesh in higher elevations, because it would always be in the distance since my game is happening at ocean level.

Question is, where I would to have to dig into to not have vertices at a given y range being optimised.

**Comment by Whinarn** [comment]
Hi @klauskobald, sorry for not getting back to you faster. I have been in the process of moving and have not had access to a computer for a few days.

The way I see it, you have two choices:

1. Implement logic of ignoring edges containing vertices at a specific elevation range in `RemoveVertexPass`  around [here](https://github.com/Whinarn/UnityMeshSimplifier/blob/81eca0b867221b745cff9c9d79a5e7b1110a36ef/Runtime/MeshSimplifier.cs#L797-L814).
2. Implement logic for calculating the error of an edge differently in `CalculateError` [here](https://github.com/Whinarn/UnityMeshSimplifier/blob/81eca0b867221b745cff9c9d79a5e7b1110a36ef/Runtime/MeshSimplifier.cs#L486).

There is unfortunately no built-in support for what you are trying to achieve. So it will require changes to the code.
Have in mind that ignoring edges will lead to no optimizations at all, so the calculation of the error would be the best way to implement a sort of prioritization of optimizations. I would recommend you to test different approaches and see what works best for you.

In short the optimization works by optimizing the edges with the lowest errors, and work it's way up until it has reached the desired triangle count, or there's nothing left to optimize (essentially if the rest of the edges are ignored). So increasing the error when the vertices are closer to your specific elevation should lead to those edges being collapsed later, hopefully leaving them mostly untouched.

I hope this helps. Let me know how this works out for you!

**Comment by klauskobald** [comment]
Thank you! That does not sound too complicated. I will check it out in a few days!

**Comment by klauskobald** [comment]
I tried it out and it works. But then I came across the SimplificationOptions! (Did not see them before). Actually it seems sufficient to preserverCurvature. :-)
